### PR TITLE
Fix issue creation and add assignee search

### PR DESCRIPTION
**Pull request by linc01n**
There is an API for us to pull out assignee information. So we no longer need to hard-code them in `jira-users.el`

I edit the `status` to use the `status` instead of `statusCategory` so when I progress the issue the status actually change( My jira defined multiple `In Progress` statuses)

I also patched the `ticketStruct` for the Rest API JSON payload format.

ps. this is my first elisp PR. I'm not sure if I am following the standard.

### CODE REVIEW

**Comment by linc01n** [comment]
The response object of `createIssue` is not fit for `org-jira-get-issues` https://github.com/baohaojun/org-jira/pull/37/files#diff-c8c30ba933ea47137229fc7e1939bee2L805

I still haven't figured out a way to make it work.

File: jiralib.el [hunk 1]
```diff
@@ -424,6 +425,17 @@ will cache it."
           (jiralib-make-assoc-list (jiralib-call "getPriorities") 'id 'name)))
   jiralib-priority-codes-cache)
 
+(defvar jiralib-assignable-users-cache nil)
+(defun jiralib-get-assignable-users (key)
+  "Return an assoc list mapping user key and display name.
+
+This function will only ask JIRA for the list of name once, than
+will cache it."
+  (unless jiralib-assignable-users-cache
```
Comment (ahungry) [inline jiralib.el]: Caching will not work well if the project changes, and a different project has a different list of assignable users.

**Comment by ahungry** [comment]
@linc01n I had many changes merged into restapi recently, this now conflicts.  I have assignee search present, as well as the fix to ticket struct, so this PR may not be necessary now.

**Comment by linc01n** [comment]
@ahungry I am new to elisp. This is just a simple hack for me to use it. Closing this since your PR cover much more area than mine.

**Pull request closed without merging**